contracts/Origin/CertificateLogic.sol

```diff
@@ -113,7 +113,7 @@ contract CertificateLogic is CertificateInterface, CertificateSpecificContract,
     /// @notice buys a certificate
     /// @param _certificateId the id of the certificate
     function buyCertificate(uint _certificateId)
-        external
+        public
```

Comment:
Instead of making this `public` you can call `this.buyCertificates()` from inside this contract.

Reply:
If you call this method from another method in this contract, `msg.sender` will not the be the sender, but this contract. Make sure this works properly by writing different tests.